enarx-keep-sgx/src/enter.s

```diff
@@ -0,0 +1,22 @@
+# SPDX-License-Identifier: Apache-2.0
+
+	.text
+    .globl enarx_eenter
+    .type enarx_eenter, @function
+enarx_eenter:
+    push   %rbx
+
+    mov    $2,        %rax    # EENTER
+    mov    %rdi,      %rbx    # TCS Address
+    lea    aep(%rip), %rcx    # AEP
+    enclu                     # Call ENCLU[EENTER]
+
+    pop    %rbx
+    ret
```

Comment:
This is a synchronous exit, right?

Reply:
Sort of, yes.

Background is here:
https://www.felixcloutier.com/x86/eenter
https://www.felixcloutier.com/x86/eexit

When we call `ENCLU[EENTER]`, the enclave receives the next address from outside the enclave as `%rcx`. The enclave will typically store that for later. When the enclave calls `ENCLU[EEXIT]` it specifies the address to jump to, which is often the saved value from before. But it doesn't need to be.

I'm working on a flow diagram for how SGX flow control will work. It is a little complex.